Code/Engine/RendererDX11/Device/DeviceDX11.h

```diff
@@ -220,9 +241,20 @@ class EZ_RENDERERDX11_DLL ezGALDeviceDX11 : public ezGALDevice
     ezUInt32 m_uiHash;
   };
 
-  ezMap<ezUInt32, ezDynamicArray<ID3D11Resource*>, ezCompareHelper<ezUInt32>, ezLocalAllocatorWrapper> m_FreeTempResources[TempResourceType::ENUM_COUNT];
+  ezMap<ezUInt32, ezDynamicArray<TempResource>, ezCompareHelper<ezUInt32>, ezLocalAllocatorWrapper> m_FreeTempResources[TempResourceType::ENUM_COUNT];
   ezDeque<UsedTempResource, ezLocalAllocatorWrapper> m_UsedTempResources[TempResourceType::ENUM_COUNT];
 
+  struct PendingCopy
+  {
+    TempResource m_SourceResource = {};
+    ID3D11Resource* m_pDestResource = nullptr;
+    ezUInt32 m_uiDestSubResource = 0;
+    ezVec3U32 m_vDestPoint = ezVec3U32::MakeZero();
+    ezVec3U32 m_vSourceSize = ezVec3U32::MakeZero();
+  };
+
+  ezDynamicArray<PendingCopy, ezLocalAllocatorWrapper> m_PendingCopies;
```

Comment:
Just for confirmation, we can get away with not double buffering this because `BeginFramePlatform` is called before the parallel rendering / update phase, so there is no risk of race conditions?

Reply:
Correct, begin and end frame are fully synchronous with no update running in parallel.